application/controllers/user.php

```diff
@@ -18,31 +18,45 @@ public function login() {
         $this->load->library('form_validation');
         $this->load->library('session');
 
-        $this->load->view('header');
+        if ($this->input->server('REQUEST_METHOD') == 'GET') {
+            $this->load->view('header');
+            $this->load->view('login_form');
+            $this->load->view('footer');
+        }
 
         if ($this->input->server('REQUEST_METHOD') == 'POST') {
             $login_info = $this->input->post();
+            header('Content-Type: application/json');
 
             if ($this->form_validation->run('login') == false) {
-                redirect(site_url('user/login'));
+                $err_code = '400';
+                $data = array(
+                    'code' => $err_code,
+                    'msg' => $GLOBALS['ERR_MSG'][$err_code]
+                );
+                echo json_encode($data);
+                exit;
             }
 
             $user_info = $this->user->get_user_by_email($login_info['mail']);
+            $err_code = '200';
 
             if (!$user_info['confirmed']) {
-                redirect(site_url('user/login'));
+                $err_code = '202';
             }
-
             if ($login_info['password'] != $user_info['password']) {
-                redirect(site_url('user/login'));
+                $err_code = '401';
             }
-            $this->session->set_userdata('logged_in', true);
-            $this->session->set_userdata('school_id', $user_info['school_id']);
-            redirect(site_url('registration/index'));
-        } else {
-            $this->load->view('login_form');
+            if ($user_info == NULL) {
+                $err_code = '204';
+            }
+
+            $data = array(
```

Comment:
I am not sure I correctly understand you. I think the problem here is to make `registration/***` a private page which only logged in users can see. And this issue may be #14 . This is the only controller which can be seen only when the user is logged in.